### PR TITLE
FAT-1460 EDIFACT file import with creating of new invoice record

### PR DESCRIPTION
**Pull request by TetianaParanich**

### CODE REVIEW

File: cypress/support/fragments/data_import/action_profiles/newActionProfile.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const folioRecordTypeValue = {
 };
 
 const defaultActionProfile = {
-  name: 'autotest action profile',
+  name: '',
```
Comment (KyryloBrener) [inline cypress/support/fragments/data_import/action_profiles/newActionProfile.js:16]: You want name to be empty?

File: cypress/support/fragments/data_import/job_profiles/newJobProfile.js [hunk 2]
```diff
@@ -1,26 +1,38 @@
 import { TextField, Select, Button, Accordion, HTML, including } from '../../../../../interactors';
 import ModalSelectActionProfile from './modalSelectActionProfile';
 
-const acceptedDataType = 'MARC';
+// const acceptedDataType = 'MARC';
```
Comment (KyryloBrener) [inline cypress/support/fragments/data_import/job_profiles/newJobProfile.js]: commented code

File: cypress/support/fragments/data_import/job_profiles/newJobProfile.js [hunk 3]
```diff
@@ -1,26 +1,38 @@
...
+  acceptedType: acceptedDataType.marc,
 };
 
 const actionsButton = Button('Action');
 
 const matchButton = Button('Match');
 
 export default {
-  acceptedDataType,
+  //acceptedDataType,
```
Comment (KyryloBrener) [inline cypress/support/fragments/data_import/job_profiles/newJobProfile.js]: commented code

Comment (TetianaParanich) [reply cypress/support/fragments/data_import/job_profiles/newJobProfile.js]: I deleted it)

File: cypress/support/fragments/data_import/job_profiles/newJobProfile.js [hunk 4]
```diff
@@ -1,26 +1,38 @@
...
-const acceptedDataType = 'MARC';
+// const acceptedDataType = 'MARC';
 
-const defaultJobProfile = {
+const acceptedDataType = {
+  marc:'MARC',
+  edifact:'EDIFACT'
+};
+
+/*const defaultJobProfile = {
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/job_profiles/newJobProfile.js]: I deleted it

File: cypress/support/fragments/data_import/job_profiles/newJobProfile.js (hunk 2 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/job_profiles/newJobProfile.js]: I deleted it

File: cypress/support/fragments/data_import/fileDetails.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import invoices from '../invoices/invoices';
+import TopMenu from '../topMenu';
+
+const expectedInvoiceDate = '10/15/2021';
+const expectedInvoiceStatus = 'Open';
+const expectedInvoiceSource = 'EDI';
+
+export default {
+  checkInvoiceDetails:() => {
+    cy.do(Section().find(MultiColumnListCell(including('49072'))).perform(element => {
```
Comment (MaksimTerekhov) [inline cypress/support/fragments/data_import/fileDetails.js:11]: Are you sure about this hardcoded value? Will it work in another environment?

`including('49072')`

File: cypress/support/fragments/data_import/logs.js [hunk 6]
```diff
@@ -44,6 +44,12 @@ const checkIsItemCreated = () => {
   }));
 };
 
+const checkIsInvoiceCreated = () => {
+  cy.do(MultiColumnListCell({ row: 0, columnIndex: 7 }).perform(element => {
```
Comment (MaksimTerekhov) [inline cypress/support/fragments/data_import/logs.js:48]: You can use simpler construction:

`cy.expect(MultiColumnListCell({ row: 0, column: 'Created' }).exists())`

File: cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js [hunk 7]
```diff
@@ -35,6 +37,15 @@ const defaultMappingProfile = {
   fillProfile:''
 };
 
+const selectOrganizationByName = () => {
+  cy.do([
+    Modal('Select Organization').find(TextField({ id: 'input-record-search' })).fillIn('GOBI Library Solutions'),
+    Modal('Select Organization').find(Button('Search')).click(),
+    Modal('Select Organization').find(HTML(including('1 record found'))).exists(),
```
Comment (MaksimTerekhov) [inline cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js:44]: Move repeating interactors into constants please

File: cypress/integration/data-import/edi-import-file-for-create-invoice-record.spec.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    const jobProfile = {
+      ...newJobProfile.defaultJobProfile,
+      profileName: jobProfileName,
+      acceptedType: newJobProfile.acceptedDataType.edifact
+    };
+
+    settingsDataImport.goToJobProfiles();
+    jobProfiles.createJobProfile(jobProfile);
+    newJobProfile.linkActionProfile(actionProfile);
+    newJobProfile.clickSaveAndCloseButton();
```
Comment (MaksimTerekhov) [inline cypress/integration/data-import/edi-import-file-for-create-invoice-record.spec.js:61]: May be use shorter names? Like 

`newJobProfile.saveAndClose()`

File: cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js [hunk 9]
```diff
@@ -169,8 +180,23 @@ export default {
...
+    cy.do([
+      TextField({ name:'profile.name' }).fillIn(specialMappingProfileName),
+      Select({ name:'profile.incomingRecordType' }).choose('EDIFACT invoice'),
+      Select({ name:'profile.existingRecordType' }).choose('Invoice'),
+      TextArea({ name:'profile.description' }).fillIn(''),
+      TextField('Batch group*').fillIn('"FOLIO"'),
+      Button('Organization look-up').click()
+    ]);
+    selectOrganizationByName();
+    cy.do([TextField('Payment method*').fillIn('"Credit Card"'),
```
Comment (MaksimTerekhov) [inline cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js:198]: Use formatter or format your code manually please :)

**Review by MaksimTerekhov — APPROVED** [review]

File: cypress/support/fragments/data_import/fileDetails.js (hunk 5 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/fileDetails.js:11]: I put the value in a variable.

File: cypress/support/fragments/data_import/logs.js (hunk 6 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/logs.js:48]: Done

File: cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js (hunk 7 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js:44]: Done

File: cypress/integration/data-import/edi-import-file-for-create-invoice-record.spec.js (hunk 8 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/edi-import-file-for-create-invoice-record.spec.js:61]: Done

File: cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js (hunk 9 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js]: Done

**Pull request merged**